components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutionDiagnosticLogger.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.execution.util;
+
+import org.apache.http.client.methods.HttpPost;
+import org.wso2.carbon.identity.action.execution.ActionExecutionLogConstants;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils;
+import org.wso2.carbon.utils.DiagnosticLog;
+
+import java.util.List;
+
+/**
+ * This utility class is for generating and printing diagnostic logs related to
+ * action execution
+ */
+public class ActionExecutionDiagnosticLogger {
+
+    public void printDiagnosticLogActionInitiation(Action action) {
```

Comment:
Since we are defining the methods insider the `ActionExecutionDiagnosticLogger` we dont need repeat the `diagnostic` part again in the method name. Also `log` would be ideal instead of using `print`

Reply:
Updated in